infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/KubernetesEnvironmentFactory.java

```diff
@@ -108,10 +108,12 @@ protected KubernetesEnvironment doCreate(
     boolean isAnyIngressPresent = false;
     boolean isAnySecretPresent = false;
     for (HasMetadata object : list.getItems()) {
+      checkNotNull(object.getKind(), "Environment contains object without specified kind field");
+      checkNotNull(object.getMetadata(), "%s metadata must not be null", object.getKind());
+      checkNotNull(object.getMetadata().getName(), "%s name must not be null", object.getKind());
```

Comment:
Perhaps this message can be improved to take into account a user attempting to use `generateName`? We don't support generateName really, but that's a Che requirement, not a k8s requirement.

Reply:
I've got no better ideas at the moment, but it's something we should think about -- probably outside the scope of this PR. 

I'm not even convinced that Che will handle generateName correctly, due to how we get e.g. deployments. I'd need to think about it more.